groovy/gradle.htmlui/src/org/netbeans/modules/gradle/htmlui/HtmlJavaApplicationProjectWizard.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.netbeans.modules.gradle.htmlui;
+
+import org.netbeans.modules.gradle.spi.newproject.SimpleGradleWizardIterator;
+import org.netbeans.modules.gradle.spi.newproject.TemplateOperation;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.netbeans.api.project.Project;
+import org.netbeans.api.templates.TemplateRegistration;
+import org.netbeans.modules.gradle.api.NbGradleProject;
+import org.openide.WizardDescriptor;
+import org.openide.filesystems.FileObject;
+import org.openide.filesystems.FileUtil;
+import org.openide.util.Exceptions;
+import org.openide.util.NbBundle.Messages;
+
+@TemplateRegistration(folder="Project/Gradle", position=999, displayName="#template.htmlUIProject", iconBase="org/netbeans/modules/gradle/htmlui/DukeHTML.png", description="description.html")
+@Messages("template.htmlUIProject=Java Frontend Application")
```

Comment:
This could be quite confusing especially when this is in the main Gradle Project Wizard folder. At least if I'd like to do a Java Frontend Application, I would not think on bck2brwsr app on the first place (I'm sorry for that).

Reply:
This template is not about bck2brwsr. The main focus is [DukeScript](http://dukescript.com) - e.g. combination of the best Java VM and portable frontend rendering. What exactly is the problem? The "Run" action? If so, it can be changed to do the same thing as "Debug".

The naming is consistent with the Maven category. There is a *Java Frontend Application* archetype in the Maven category which leads to the same/similar application, using the same libraries. Our intention is to deliver the same *Java Frontend Application* experience to Gradle users.

Reply:
Btw. there is also a way to run the application as [Android application](http://kt-mvvm.org/android.html) and as [iOS application](http://kt-mvvm.org/ios.html). They aren't advertised in the UI yet as it requires additional configuration of SDK, simulator, etc. We'd welcome coordinated support in this direction of generic *Java Frontend Applications*, but we think it is out of the scope of this PR.